server/services/src/main/java/org/zanata/adapter/TSAdapter.java

```diff
@@ -196,7 +199,8 @@ protected void generateTranslatedFile(URI originalFile,
                 Event event = filter.next();
                 if (event.isDocumentPart() &&
                         event.getDocumentPart().hasProperty("language")) {
-                    // TODO ZNTA-2483 change readonly language property
+                    event.getDocumentPart().setSkeleton(
+                            replaceLocaleInDocPart(event, localeId.toString()));
```

Comment:
toBCP47 would be a bit safer, in fact why don't you pass the Okapi LocaleId into the method? That way it won't have to check the string against a regex.